data/KPI_functions.py

```diff
@@ -43,21 +43,34 @@ def heat_gen(dataDict, buildings, timeStep):
 
 
     for b in range(1, buildings+1):
-        shGenList = ["(('CHP__Building" + str(b) + "', 'shSourceBus__Building" + str(b) + "'), 'flow')",
-                     "(('GWHP__Building" + str(b) + "', 'shSourceBus__Building" + str(b) + "'), 'flow')",
-                     "(('GasBoiler__Building" + str(b) + "', 'shSourceBus__Building" + str(b) + "'), 'flow')",
-                     "(('HP__Building" + str(b) + "', 'shSourceBus__Building" + str(b) + "'), 'flow')",
-                     "(('ElectricRod__Building" + str(b) + "', 'shSourceBus__Building" + str(b) + "'), 'flow')"]
-
-        dhwGenList = ["(('CHP__Building" + str(b) + "', 'dhwStorageBus__Building" + str(b) + "'), 'flow')",
-                      "(('GWHP__Building" + str(b) + "', 'dhwStorageBus__Building" + str(b) + "'), 'flow')",
-                      "(('GasBoiler__Building" + str(b) + "', 'dhwStorageBus__Building" + str(b) + "'), 'flow')",
-                      "(('HP__Building" + str(b) + "', 'dhwStorageBus__Building" + str(b) + "'), 'flow')",
-                      "(('ElectricRod__Building" + str(b) + "', 'dhwStorageBus__Building" + str(b) + "'), 'flow')",
-                      "(('solarCollector__Building" + str(b) + "', 'dhwStorageBus__Building" + str(b) + "'), 'flow')"]
-
-        SHUnitIndex = ['CHPsh', "GWHPsh", "GasBoilersh", "HPsh", "ElectricRodsh"]
-        DHWUnitIndex = ['CHPdhw', "GWHPdhw", "GasBoilerdhw", "HPdhw", "ElectricRoddhw", "solarCollectordhw"]
+        if "(('ElectricRod__Building" + str(b) + "', 'shSourceBus__Building" + str(b) + "'), 'flow')" in list(dataDict["shSourceBus__Building" + str(b)]):
```

Comment:
Why is this added conditionally? Do the lists 'shGenList' and 'dhwGenList' need only those technologies which are possible? If yes, then what happens if we have only CHP and ASHP technologies active but the rest are inactive? Would the script still run?